docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-conditions-autodiscover.asciidoc

```diff
@@ -0,0 +1,314 @@
+[[conditions-based-autodiscover]]
+= Conditions based autodiscover
+
+You can define autodiscover conditions in each input to allow {agent} to automatically identify Pods and start monitoring them using predefined integrations. You can find more about these inputs and their configurations <<elastic-agent-input-configuration,here>>.
+
+== Example: Target pods by host and label
+
+To automatically identify a Redis Pod and monitor it with the Redis integration, uncomment the following input configuration in the {agent} manifest:
+
+
+[source,yaml]
+------------------------------------------------
+- name: redis
+  type: redis/metrics
+  use_output: default
+  meta:
+    package:
+      name: redis
+      version: 0.3.6
+  data_stream:
+    namespace: default
+  streams:
+    - data_stream:
+        dataset: redis.info
+        type: metrics
+      metricsets:
+        - info
+      hosts:
+        - '${kubernetes.pod.ip}:6379'
+      idle_timeout: 20s
+      maxconn: 10
+      network: tcp
+      period: 10s
+      condition: ${kubernetes.labels.app} == 'redis'
+------------------------------------------------
+
+The condition `${kubernetes.labels.app} == 'redis'` will make the {agent} look for a Pod with the `redis` label app within the scope defined in its manifest. You can find the exported fields of the Kubernetes integration {metricbeat-ref}/exported-fields-kubernetes-processor.html[here].
```

Comment:
@dedemorton I dont like here that we point to metricbeat links. Can we change that?

Also for kubernetes I can see two pages created, one for metadata and one for processor because they come from different places but with same title

Reply:
@gizas Yah that's not ideal. I'm looking at your comment a bit out of context, but I think it's also strange that the link says the fields are added by the K8s integration, but the link points to the fields added by the processor. Does the integration (or provider) call the processor under the hood? Should we be pointing to the field reference in the [integrations documentation](https://docs.elastic.co/integrations)? Sorry to answer your question with more questions, but I haven't had a chance to dig into this problem very deeply.

I'm actually working on adding processor docs to the Fleet/Agent/Integrations documentation, but I haven't started documenting processors like `add_kubernetes_metadata` because I don't know if they are handled by the provider implementation. If we _do_ include those metadata processors, we will show examples of the events returned, but TBH, I worry about the content getting stale if it's outside of the process used to generate reference content from yaml files. And I think it's not a great user experience to jump around to find this information. Some kind of consolidated reference of all the fields with an explanation of how they get added would probably be a lot more useful to users.

Do you want to open a separate issue to track this problem?

Reply:
I guess we can replace above link with below resources that the provider gives:
https://www.elastic.co/guide/en/fleet/master/kubernetes-provider.html#_provider_for_pod_resources
https://www.elastic.co/guide/en/fleet/master/kubernetes-provider.html#_provider_for_service_resources
https://www.elastic.co/guide/en/fleet/master/kubernetes-provider.html#_provider_for_node_resources

Indeed I did a little search and could not find a reference for processors in Agent. In our case the processor is handled by agent under the hood.